### ISSUE TITLE
How do I parse .dump files?

### ISSUE THREAD

**Issue by TheOnlyWayUp** [PrismarineJS/minecraft-chunk-dumper#27]
My goal is to convert a .dump file to an array of block IDs, going from the lowest Y to the height limit. I'm doing this in Python and I'm not sure how the file is encoded, nor am I sure how I can decode it, the amount of bytes is much more than the height limit too, I've also gone through [Prismarine-Chunk](https://github.com/PrismarineJS/prismarine-chunk/blob/d9eca7174fda9fa7ca7d1a95aec01d21f2771495/src/pc/1.8/chunk.js#L213) but I didn't understand what it did, if I can't parse it into block IDs, is there a way I can generate a top view image from it? Thanks!

TL:DR - How do I parse .dump files into an array of block IDs?

**Comment by rom1504** [comment]
see https://wiki.vg/Chunk_Format https://github.com/PrismarineJS/prismarine-chunk/ 
some other implementations there https://wiki.vg/Chunk_Format#Full_implementations
including a python one https://github.com/barneygale/quarry/blob/master/quarry/types/chunk.py

you can also use https://github.com/extremeheat/JSPyBridge also with prismarine-chunk to stay in python while having all versions support

chunks in minecraft are the most complicated part of the serialization. There is no easy way out of this
Either you need to take the time to understand it, either you use an existing lib

**Comment by TheOnlyWayUp** [comment]
Interesting, thank you so much for the fast response and the useful resources! I'm going over them right now, I'd like to know though, is there a way I can directly integrate these .dump files into the [Prismarine Viewer](https://github.com/PrismarineJS/prismarine-viewer)?

**Comment by rom1504** [comment]
I guess you could read them with pchunk, put them in a pworld then use something like https://github.com/PrismarineJS/prismarine-world/blob/master/examples/viewer/index.js

what is your goal ? 

I figure https://github.com/PrismarineJS/mineflayer/blob/master/examples/anvil_saver/saver.js + https://github.com/PrismarineJS/prismarine-world/blob/master/examples/viewer/index.js might be closer to what you want (saving anvil with mineflayer then viewing with pviewer)

**Comment by TheOnlyWayUp** [comment]
My goal is to join a server, download some chunks and display that on my website just as a proof of concept, does Mineflayer not have a limited range of versions? And, it's also slightly complicated. I'll definitely look into using Mineflayer. Thanks for your quick responses, I'd like to know, is there a better way to ask you questions?

**Comment by rom1504** [comment]
> does Mineflayer not have a limited range of versions

yes 1.8 to 1.17

> And, it's also slightly complicated. I'll definitely look into using Mineflayer

https://github.com/PrismarineJS/mineflayer/blob/master/examples/anvil_saver/saver.js if you run that it gets you a world save of that server

> is there a better way to ask you questions?

you can join our discord
https://discord.gg/GsEFRM8
